jobbergate-api/jobbergate_api/apps/services.py

```diff
@@ -281,7 +307,42 @@ async def list(self, **filter_kwargs) -> list[CrudModel]:
         For details on the supported filters, see the ``build_list_query()`` method.
         """
         result: Result = await self.session.execute(self.build_list_query(**filter_kwargs))
-        return list(result.scalars())
+        return list(result.unique().scalars())  # type: ignore
+
+    @asynccontextmanager
+    async def ensure_attribute(self, locator: Any, **attributes):
```

Comment:
Why make this a context manager? I'm not seeing that there is any setup/teardown involved. It seems like it's executing an extra `select` query just to check for the attributes. Why couldn't it be an extension of the `get()` method that checks for the attributes after fetching the instance?

Reply:
solved in [864a86d](https://github.com/omnivector-solutions/jobbergate/pull/292/commits/864a86dadd9a9a2c82b002b743bdc9f4add2cd15)